mars/deploy/oscar/tests/test_ray_dag.py

```diff
@@ -129,3 +159,131 @@ async def test_session_get_progress(ray_start_regular_shared2, create_cluster):
+        n_cpu=2,
+        web=False,
+        use_uvloop=False,
+        config={"task.execution_config.ray.subtask_monitor_interval": 0},
+    )
+
+    assert session._session.client.web_address is None
+    assert session.get_web_endpoint() is None
+
+    def f1(c):
+        time.sleep(0.5)
+        return c
+
+    with session:
+        t1 = mt.random.randint(10, size=(100, 10), chunk_size=100)
+        t2 = mt.random.randint(10, size=(100, 10), chunk_size=50)
+        t3 = t2 + t1
+        t4 = t3.sum(0)
+        t5 = t4.map_chunk(f1)
+        r = t5.execute()
+        result = r.fetch()
+        assert result is not None
+        assert len(result) == 10
+        context = get_context()
+        assert len(context._task_context) < 5
+
+    session.stop_server()
+    assert get_default_async_session() is None
+
+
+class MockRayTaskExecutor(RayTaskExecutor):
+    async def submit_subtask_graph(
+        self,
+        stage_id: str,
+        subtask_graph: SubtaskGraph,
+        chunk_graph: ChunkGraph,
+    ):
+        monitor_task = asyncio.create_task(
+            self._update_progress_and_collect_garbage(
+                subtask_graph, self._config.get_subtask_monitor_interval()
+            )
+        )
+
+        result_meta_keys = {
+            chunk.key
+            for chunk in chunk_graph.result_chunks
+            if not isinstance(chunk.op, Fetch)
+        }
+
+        for subtask in subtask_graph.topological_iter():
+            subtask_chunk_graph = subtask.chunk_graph
+            task_context = self._task_context
+            key_to_input = await self._load_subtask_inputs(
+                stage_id, subtask, subtask_chunk_graph, task_context
+            )
+            output_keys = self._get_subtask_output_keys(subtask_chunk_graph)
+            output_meta_keys = result_meta_keys & output_keys
+            output_count = len(output_keys) + bool(output_meta_keys)
+            output_object_refs = self._ray_executor.options(
+                num_returns=output_count
+            ).remote(
+                subtask.task_id,
+                subtask.subtask_id,
+                serialize(subtask_chunk_graph),
+                output_meta_keys,
+                list(key_to_input.keys()),
+                *key_to_input.values(),
+            )
+            if output_count == 0:
+                continue
+            elif output_count == 1:
+                output_object_refs = [output_object_refs]
+            self._cur_stage_first_output_object_ref_to_subtask[
+                output_object_refs[0]
+            ] = subtask
+            if output_meta_keys:
+                meta_object_ref, *output_object_refs = output_object_refs
+            task_context.update(zip(output_keys, output_object_refs))
+
+        return monitor_task
+
+
+class MockTileContext(TileContext):
+    def get_all_progress(self) -> float:
+        return 1.0
+
+
+@require_ray
+@pytest.mark.asyncio
+async def test_detail_context_gc():
```

Comment:
This test case can be moved to [mars/services/task/execution/ray/tests/test_ray_execution_backend.py](https://github.com/mars-project/mars/pull/3061/files#diff-1b9b2920089480060025101707a9fb8efe124b5084de87888b2e38cc944eb299) because no deployment is needed.

Reply:
OK.